### PR TITLE
Adapt to `__future__.annotations` in `optuna/storages/_rdb/models.py`

### PR DESCRIPTION
**Pull request by aisha-partha**
## Motivation
This is my first contribution to Optuna. I would like to contribute to the resolution of https://github.com/optuna/optuna/issues/4508

## Description of the changes
Added
__future__.annotations

Deleted:
typing.List
typing.Tuple
typing.Optional

### CODE REVIEW

File: optuna/storages/_rdb/models.py [hunk 1]
```diff
@@ -261,8 +260,8 @@ def find_or_raise_by_id(
     def count(
         cls,
         session: orm.Session,
-        study: Optional[StudyModel] = None,
-        state: Optional[TrialState] = None,
+        study: StudyModel | None = None,
+        state: TrialState | None = None,
```
Comment (nabenabe0928) [inline optuna/storages/_rdb/models.py:264]: Could you modify it to the following?

```python
    @classmethod
    def count(
        cls, session: orm.Session, study: StudyModel | None = None, state: TrialState | None = None
    ) -> int:
```

File: optuna/storages/_rdb/models.py [hunk 2]
```diff
@@ -431,7 +430,7 @@ class TrialValueType(enum.Enum):
     def value_to_stored_repr(
         cls,
         value: float,
-    ) -> Tuple[Optional[float], TrialValueType]:
+    ) -> tuple[float | None, TrialValueType]:
```
Comment (nabenabe0928) [inline optuna/storages/_rdb/models.py:433]: Could you modify it to the following?
```python
    @classmethod
    def value_to_stored_repr(cls, value: float) -> tuple[float | None, TrialValueType]:
```

File: optuna/storages/_rdb/models.py [hunk 3]
```diff
@@ -497,7 +496,7 @@ class TrialIntermediateValueType(enum.Enum):
     def intermediate_value_to_stored_repr(
         cls,
         value: float,
```
Comment (nabenabe0928) [inline optuna/storages/_rdb/models.py:498]:
```suggestion
        cls, value: float
```

**Review by nabenabe0928 — CHANGES_REQUESTED** [review]
Thank you for your PR!
I left some comments!

File: optuna/storages/_rdb/models.py (hunk 3 above)
Comment (aisha-partha) [reply optuna/storages/_rdb/models.py]: The changes have been made as suggested.

**Review by nabenabe0928 — APPROVED** [review]
LGTM!

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**